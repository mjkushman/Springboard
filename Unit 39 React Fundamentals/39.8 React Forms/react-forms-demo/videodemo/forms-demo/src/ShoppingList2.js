import React, { useState } from "react";
import Item from "./Item";
import NewItemForm2 from "./NewItemForm2";
import {v4 as uuid} from 'uuid';

const ShoppingList = () => {

  const INITIAL_STATE = [
    {id:1, name:'Peanuts', qty: 4, id:uuid()},
    {id:2, name:'Salami', qty: 3, id:uuid()}
  ]
  const [items, setItems] =useState(INITIAL_STATE)

  const addItem = (newItem) => {
    setItems(items => [...items, { ...newItem, id: uuid() }]) // makes a new array of items that includes the existing items plus new item spread out. And add uuid
  } 

  
  return (
  <div>
    <h3>Shopping List</h3>
    <div><NewItemForm2 addItem={addItem} /></div>
    <div>
      {items.map(({id, name, qty }) => <Item id={id} qty={qty} name={name} key={id}/>)}
    </div>
  </div>
  )

}

export default ShoppingList;


