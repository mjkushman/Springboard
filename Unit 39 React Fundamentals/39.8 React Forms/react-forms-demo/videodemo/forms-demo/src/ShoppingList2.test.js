import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ShoppingList2 from "./ShoppingList2";

//Smoke test
it("renders without crashing", function () {
  render(<ShoppingList2 />);
});

//snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(<ShoppingList2 />);
  expect(asFragment()).toMatchSnapshot();
});


it("should add new item", function() {
  const {queryByText, getByLabelText} = render(<ShoppingList2/>);
  const input = getByLabelText("Product")
  const btn = queryByText("Add Item")
  
  // make sure Pizza is not in the document
  expect(queryByText('Product Name: Pizza')).not.toBeInTheDocument();

  // mock fill out the form
  fireEvent.change(input, {target:{value: 'Pizza'} } )
  
  //click the submit button
  fireEvent.click(btn)

  // now, expect pizza to be in the DOM
  expect(queryByText('Product Name: Pizza')).toBeInTheDocument();
})