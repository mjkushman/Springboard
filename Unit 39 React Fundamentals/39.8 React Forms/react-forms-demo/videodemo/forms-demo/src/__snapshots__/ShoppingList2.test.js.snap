// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Shopping List
    </h3>
    <div>
      <form>
        <label
          for="name"
        >
          Product
        </label>
        <input
          id="name"
          name="name"
          placeholder="Item Name"
          type="text"
          value=""
        />
        <label
          for="qty"
        >
          Quantity: 0
        </label>
        <input
          id="qty"
          max="10"
          min="0"
          name="qty"
          type="range"
          value="0"
        />
        <button>
          Add Item
        </button>
      </form>
    </div>
    <div>
      <ul>
        <li>
          Product Name: Peanuts
        </li>
        <li>
          Quantity: 4
        </li>
      </ul>
      <ul>
        <li>
          Product Name: Salami
        </li>
        <li>
          Quantity: 3
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
